librepo/handle.c

```diff
@@ -264,6 +266,32 @@ lr_handle_remote_sources_changed(LrHandle *handle, LrChangedRemoteSource type)
     }
 }
 
+struct callback_data {
+   GMainLoop *loop; 
+   long input_seconds;
+   time_t begin;
+   GNetworkMonitor *monitor;
+   GSocketConnectable *connectable; 
+   GCancellable *cancellable;
+};
+
+gboolean timeout_callback(gpointer data)
+{ 
+  struct callback_data *dt = (struct callback_data*)data;
+   if(!g_network_monitor_get_network_available(dt->monitor) &&
```

Comment:
One thing I think we should do here is have the availability of the network control whether or not we're polling.  IOW, something like:

```
  if (!g_network_monitor_get_network_available(dt->monitor))
      return FALSE;
```